### DISCUSSION TITLE
Livewire v3 - how to restrict child component to dispatch only to its immediate parent and not beyond?

### DISCUSSION THREAD

**Discussion by akulmehta** [livewire/livewire#7969] [1. Help]
On a page I have two parent components `ParentA_1` and `ParentA_2` both are of type `ParentA`. Both parent components can have several `Child` components.

The `Child` component has a delete button which when pressed used to `emitUp` (Livewire v2) to the corresponding parent component to which it belonged to delete it from that parent component and it would be scoped to the parent component. In v3 however, `emitUp` has been replaced by `dispatch` and now when the delete event is dispatched, it hits both parent components `ParentA_1` and `ParentA_2`.

This causes some errors like 
```
Livewire\Features\SupportLegacyModels\EloquentModelSynth::setDataOnModel(): Argument #1 ($model) must be of type Illuminate\Database\Eloquent\Model, null given, called in ...\vendor\livewire\livewire\src\Features\SupportLegacyModels\EloquentModelSynth.php on line 89
```

How would one go about restricting the `dispatch` to only the parent component?

**Comment by danie-ramdhani** [comment]
have you tried `dispatch()->to()` ? (https://livewire.laravel.com/docs/events#dispatching-directly-to-another-component)

```php
        $this->dispatch('delete')->to(ParentA::class);
```

the best one is using `$parent` (https://livewire.laravel.com/docs/actions#parent)

```blade
<button wire:click="$parent.removePost({{ $post->id }})">Remove</button>
```

**Reply by akulmehta** [reply]
I am sorry - I made a mistake. Both parents are of the same component. I have updated my OP above.

**Comment by joshhanley** [comment]
@akulmehta you can listen for an event directly on a child, by doing

```blade
<livewire:child @delete="removePost($event.detail.id)" />
```

See here for more details https://livewire.laravel.com/docs/events#listening-for-events-from-specific-child-components

**Reply by akulmehta** [reply]
@joshhanley Thanks. Is it possible to set the `@` dynamically? 

Something like:

```html
<livewire:edit-post @{{$dispatchEventName}}="close">
```

**Reply by joshhanley** [reply]
@akulmehta hmm not sure. I don't know if Blade supports that. If it does then it should work. But I suspect the blade renderer will have issues.

**Comment by lgmworks** [comment]
You can listen in blade in the parent and stop propagation eg:
` @saved.stop="handleSaved($event.detail.fieldName, $event.detail.fieldData)"`
